src/aspire/source/relion.py

```diff
@@ -164,11 +166,21 @@ def __init__(
             # filter_indices stores, for each particle index, the index in
             # self.unique_filters of the filter that should be applied
             self.filter_indices = filter_indices
+
+        # We have provided some, but not all the required params
+        elif any(param in metadata.columns for param in CTF_params):
```

Comment:
Shouldn't this branch raise an error instead of a warning, since under this condition we won't have any `CTFFilter`s but it also won't ever add the default `IdentityFilter`?